filebeat/beater/filebeat.go

```diff
@@ -96,23 +105,47 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 	defer spooler.Stop()
 	defer publisherChan.Close()
 
-	err = crawler.Start(registrar.GetStates())
+	err = crawler.Start(registrar.GetStates(), *once)
 	if err != nil {
 		return err
 	}
 	// Stop crawler -> stop prospectors -> stop harvesters
 	defer crawler.Stop()
 
-	// Blocks progressing. As soon as channel is closed, all defer statements come into play
+	// Wait for completion if running once.
+	if *once {
+		go func() {
+			logp.Info("Running filebeat once. Waiting for completion ...")
+			crawler.WaitForCompletion()
+
+			// Flushes spooler as often as needed to send all events to publisher
+			for spooler.Flush() > 0 {
+			}
+
+			// Wait for registrar to finish writing registry
+			finishedWg.Wait()
+			logp.Info("All data collection completed. Shutting down.")
```

Comment:
this code is already run on shutdown after Kill signal being received (the crawler is closed before being finished). So what todo here? wait for completion of require CTRL-C to kill the process as is instead of waiting for all data being published. Wonder if introducing stop timeout would mitigate this issue.